source/appModules/teamtalk5classic.py

```diff
@@ -0,0 +1,245 @@
+LBL_VIDEO = _("Video")
+# Translators: Status text in the TeamTalk voice conferencing client.
+# Note: this string is slightly different from the actual toolbar text
+# to avoid a translatable string conflict.
+LBL_DESKTOP = pgettext("TeamTalk", "Desktop")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_STREAM = _("Stream To Channel")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_MUTE = _("Mute All")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_RECORD = _("Store Audio")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_MSG = _("Show Channel Messages")
+
+#: The names, in order, of the toolbar items.
+#: The count must match exactly or this module will not function.
+#: The name "sep" is a placeholder for separators in the toolbar.
+#: These are name sets per client version.
+#: Stored in reverse so current clients find the right list faster.
+_msgNames = {
+	"4.5": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		LBL_VIDEO,
+		LBL_DESKTOP,
+		LBL_STREAM,
+		"sep",
+		LBL_MUTE,
+		LBL_RECORD,
+		"sep",
+		LBL_MSG,
+	],
+	"4.3": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		LBL_VIDEO,
+		LBL_DESKTOP,
+		LBL_MUTE,
+		LBL_RECORD,
+		"sep",
+		LBL_MSG,
+	],
+	"4.2": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		"sep",
+		LBL_MUTE,
+		LBL_RECORD,
+		"sep",
+		LBL_MSG,
+	],
+	"4.1": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		"sep",
+		LBL_MUTE,
+		"sep",
+		LBL_MSG,
+	],
+}
+
+# window control IDs
+ID_CHATOUT = 1204
+ID_CHATIN = 1206
+ID_BANNEDLIST = 1326
+ID_UNBANNEDLIST = 1327
+ID_TOOLBAR = 59392
+ID_VU = 1004
+
+
+class TTToolBarInfo(object):
+	"""Info about one TeamTalk toolbar and its items.
+	"""
+
+	def __init__(self, NVDAObj=None):
+		if not NVDAObj:
+			# Use the IAccessible for the active TeamTalk instance's toolbar.
+			NVDAObj = self.getToolBar()
+			if not NVDAObj:
+				# Translators: Reported when status information is unavailable
+				# in the TeamTalk voice conferencing client.
+				ui.message(_("Status unavailable"))
+				return
+		self.NVDAObj = NVDAObj
+		n = NVDAObj.childCount
+		# TeamTalk version-specific toolbar item names by toolbar item count.
+		self.names = []
+		for lst in _msgNames.values():
+			if len(lst) == n:
+				self.names = lst
+				break
+
+	def getToolBar(self):
+		# First lastChild is a Window object, then the client toolbar object.
+		try:
+			tb = api.getForegroundObject().lastChild.lastChild
+		except AttributeError:
+			tb = None
+		return tb
+
+	def name(self, itemID):
+		"""The name of one item.
+		"""
+		if itemID <= 0:
+			return ""
+		# ChildIDs are 1-based, but Python arrays are 0-based.
+		try:
+			name = self.names[itemID - 1]
+		except IndexError:
+			return ""
+		if name == "sep":
+			return ""
+		return name
+
+
+class TTToolBar(IAccessible):
+	def initOverlayClass(self):
+		self.info = TTToolBarInfo(self)
```

Comment:
If I understand the code correctly, TTToolBarInfo gets the toolbar itself when it is not initialized with an object. However, on that toolbar object, the overlay will be added and another TTToolBarInfo will be created. I think this code is too complicated. Why not add all the logic from TTToolBarInfo to this overlay instead?